### PR TITLE
PLAT-80499: Transition to external iLib NPM module

### PR DESCRIPTION
**Pull request by JayCanuck**
Requires: https://github.com/enactjs/dev-utils/pull/44
Once that and https://github.com/enactjs/templates/pull/11 are merged, this PR can be updated for their new dependency releases.

Updates the Webpack and Jest configs for the new iLib location, along with temporary backwards-compatibility alias support so `@enact/i18n/ilib/...` imports/require statements still work (allowing app devs time to update).

Enact-DCO-1.0-Signed-off-by: Jason Robitaille <jason.robitaille@lge.com>

### CODE REVIEW

File: config/webpack.config.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ module.exports = function(env) {
 	// Load applicable .env files into environment variables.
 	require('./dotenv').load(app.context);
 
+	// Determine dynamic iLib location for aliasing
+	const iLib = ['ilib', '@enact/i18n/node_modules/ilib'].find(f =>
```
Comment (ryanjduffy) [inline config/webpack.config.js]: Should we use the same list of directories as in the jest config? Also, I'm curious when `@enact/i18n/node_modules/ilib` would be valid.

Comment (JayCanuck) [reply config/webpack.config.js]: The jest config allows for both compatibility aliases simulataeously, whereas webpsack does not. Plus Jest includes code to set the `ILIB_BASE_PATH` env var and handle in-library tests, so more verbose detection is needed there, whereas in webpack we can reduce to a simple '@enact/i18n/ilib` path and let webpack do the resolving.

As for `@enact/i18n/node_modules/ilib`, I think you're right and under the peer dependency model, we should be able to drop it.  Any symlink scenario would still reflect peer dependency and require the top-level app to have an `ilib` dependency.

Comment (ryanjduffy) [reply config/webpack.config.js]: Just occurred to me that we _might_ need it for unit testing.

Comment (JayCanuck) [reply config/webpack.config.js]: Hmm, for jest, moonstone tests and i18n tests both have devDependency of `ilib`, so I believe they should still resolve to `node_modules/ilib`

Comment (ryanjduffy) [reply config/webpack.config.js]: Perfect. That's best.

File: config/webpack.config.js [hunk 2]
```diff
@@ -154,10 +154,11 @@ module.exports = function(env) {
...
 			modules: [path.resolve('./node_modules'), 'node_modules'],
-			alias: {
-				// Support ilib shorthand alias for ilib modules
-				ilib: '@enact/i18n/ilib/lib'
-			}
+			// Backward compatibility for apps using new ilib references with old Enact
+			// and old apps referencing old iLib location with new Enact
+			alias: !fs.existsSync(path.join(app.context, 'node_modules', 'ilib'))
+				? {ilib: '@enact/i18n/ilib'}
+				: {'@enact/i18n/ilib': 'ilib'}
```
Comment (webOS101) [inline config/webpack.config.js]: :neat:

File: config/jest/jest.config.js [hunk 3]
```diff
@@ -18,9 +19,10 @@ process.env.PUBLIC_URL = '';
 process.env.BROWSERSLIST = 'current node';
 
 const pkg = packageRoot();
+const iLibDirs = ['node_modules/@enact/i18n/ilib', 'node_modules/ilib', 'ilib'];
 const globals = {
 	__DEV__: true,
-	ILIB_BASE_PATH: 'node_modules/@enact/i18n/ilib',
+	ILIB_BASE_PATH: iLibDirs.find(f => fs.existsSync(path.join(pkg.path, f))) || iLibDirs[0],
```
Comment (webOS101) [inline config/jest/jest.config.js]: Can you explain why you chose to use the old location as the fallback in case none exist?

Comment (JayCanuck) [reply config/jest/jest.config.js]: Whoops, that's a remnant of the change earlier today. Fallback should be tthe new location (and when a fallback, primarily use for global value usage. I'll update

**Pull request merged**